rust-connectors/sources/postgres/src/model.rs

```diff
@@ -0,0 +1,451 @@
+use serde::{Deserialize, Serialize};
+
+/// One top-level event from the Postgres logical replication stream.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct ReplicationEvent {
+    /// THe Postgres WAL offset before this event.
+    pub wal_start: u64,
+    /// THe Postgres WAL offset after this event.
+    pub wal_end: u64,
+    /// The timestamp when this event occurred.
+    pub timestamp: i64,
+    /// The payload of the event, describing what took place.
+    pub message: LogicalReplicationMessage,
+}
+
+/// The payload types of a logical replication event.
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(tag = "type", rename_all = "lowercase")]
+pub enum LogicalReplicationMessage {
+    /// The beginning of a transaction.
+    Begin(BeginBody),
+    /// The ending of a transaction that has been successfully committed.
+    Commit(CommitBody),
+    /// Indicates that this transaction began on a different replication node.
+    Origin(OriginBody),
+    /// Describes the schema of a relation (e.g. table).
+    Relation(RelationBody),
+    /// A Type replication message
+    Type(TypeBody),
+    /// A new row was inserted to the specified relation with the contained data.
+    Insert(InsertBody),
+    /// A row was updated in the specified relation with the contained data.
+    Update(UpdateBody),
+    /// A row was deleted in the specified relation with the contained data and key.
+    Delete(DeleteBody),
+    /// The given relations were truncated
+    Truncate(TruncateBody),
+}
+
+/// A row as it appears in the replication stream
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Tuple(pub Vec<TupleData>);
+
+/// A column as it appears in the replication stream
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Column {
+    /// Currently may be 0 or 1, representing whether this column is part of the key.
+    pub flags: i8,
```

Comment:
We might be able to do something such as create an enum like `ColumnFlags`, but I think it may be subject to change over time, within the bounds of an i8, which is why I left it as-is.

This is a direct mapping to the Postgres logical replication protocol: https://www.postgresql.org/docs/14/protocol-logicalrep-message-formats.html